Sup 46 warehouse page

alexeyoverchenko opened the pull request:
- warehouse page

alexeyoverchenko committed:
- in progress

alexeyoverchenko committed:
-

alexeyoverchenko committed:
-modal edit

alexeyoverchenko committed:
-added handleCountPerPage

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

alexeyoverchenko committed:
-working with edit modal

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

alexeyoverchenko committed:
-button for type

alexeyoverchenko committed:
- working state list
- fixed perPage and changePage bugs

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/App.js
#	frontend/src/Header.js
#	src/main/java/by/itech/lab/supplier/constant/ApiConstants.java

alexeyoverchenko committed:
-merged with develop

alexeyoverchenko committed:
- added dropdown menu into the ModalAddWarehouse.js
- added "search state" menu into the ModalAddWarehouse.js

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/App.js
#	frontend/src/Header.js

alexeyoverchenko committed:
- checkbox in progress

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/App.js
#	frontend/src/Header.js
#	src/main/java/by/itech/lab/supplier/controller/pre_tests.http
#	src/main/java/by/itech/lab/supplier/repository/StateRepository.java
#	src/main/java/by/itech/lab/supplier/service/StateService.java
#	src/main/java/by/itech/lab/supplier/service/impl/StateServiceImpl.java

alexeyoverchenko committed:
- merge with develop

alexeyoverchenko committed:
- added validation into the ModalEditWarehouse.js

alexeyoverchenko committed:
- in progress

alexeyoverchenko committed:
- finished with checkbox for delete

alexeyoverchenko committed:
- finished with validation

alexeyoverchenko committed:
- add and delete users field in ModalAddWarehouse.js

alexeyoverchenko committed:
- added dispatcher-edit functional in ModalEditWarehouse.js

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	frontend/package.json
#	frontend/src/Header.js
#	src/main/java/by/itech/lab/supplier/constant/ApiConstants.java
#	src/main/java/by/itech/lab/supplier/controller/UserController.java
#	src/main/java/by/itech/lab/supplier/controller/pre_tests.http
#	src/main/java/by/itech/lab/supplier/service/UserService.java

alexeyoverchenko committed:
- after merge

alexeyoverchenko committed:
- clean validation

alexeyoverchenko committed:
- edited and added validation

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	src/main/java/by/itech/lab/supplier/constant/ApiConstants.java
#	src/main/java/by/itech/lab/supplier/controller/WarehouseController.java
#	src/main/java/by/itech/lab/supplier/controller/pre_tests.http

eugenmolchanov commented on frontend/src/App.js:
```diff
@@ -35,6 +36,12 @@ function App() {
     };
 
+    const renderWarehouse = () => {
+        return <ProtectedComponent conditions={user} render={(() => {
```
Is that enough condition in conditions prop?

alexeyoverchenko replied:
you're right. it's not enough

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+    const ref = React.createRef();
+    const [stateOptions, setStateOptions] = useState([]);
+    const [dispatcherOptions, setDispatcherOptions] = useState([]);
+    const [dispatchersList, setDispatchersList] = useState([]);
```
Rename dispatchersList to dispatchers. Take it as a general rule for list variables naming.

alexeyoverchenko replied:
thanks

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+        fetch(`/customers/${props.currentCustomerId}/users/dispatchers?username=${query}`)
+            .then(resp => resp.json())
+            .then(res => {
+                const optionsFromBack = res.map((i) => ({
```
Give i a reasonable name.

alexeyoverchenko replied:
got it

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+                ...preState,
+                addressDto: {...preState.addressDto, state: {id: '', state: ''}}
+            }));
+    };
```
Put e.length > 0 and ternary operator to the place where these to settings are different (state setting.). So you'll have one setWareHouseDto statement.

alexeyoverchenko replied:
I don't have the other place to set this field... should I create it?

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+        if (e.length !== 0) {
+            const dispatcherId = e[0].id;
+            const isContains = dispatchersList.filter((dispatcher) => dispatcher.id === dispatcherId);
+            if (isContains.length === 0) {
```
You may use find function instead of filter.

alexeyoverchenko replied:
thanks

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+            validationErrors: validationResult,
+            serverErrors: ''
+        }));
+        if (validationResult.length === 0) {
```
if (!validationResult.length) may be used. Read about boolean evaluation in js.

alexeyoverchenko replied:
cool, I'll take it

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+
+    const showDispatchers = dispatchersList.map(disp =>
+        <div key={disp.id}>
+            {disp.name + " " + disp.surname + ", username: " + disp.username}
```
Use es6 string concatenation.

alexeyoverchenko replied:
got it

alexeyoverchenko replied:
between div scope I can't do something like `My name is ${person.name}`, but I can use 
div {disp.name} {disp.surname}, username: {disp.username} /div
it's ok? I think, it seems clearer than it was before

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+    return (
+        <>
+            <Modal
+                show={props.props}
```
props.props seems wrong.

alexeyoverchenko replied:
agree

eugenmolchanov commented on frontend/src/pages/warehouse/ModalAddWarehouse.js:
```diff
@@ -0,0 +1,324 @@
+                                    {dropdownMenuName}
+                                </Dropdown.Toggle>
+                                <Dropdown.Menu>
+                                    <Dropdown.Item onClick={() => handleType("FACTORY")}>FACTORY</Dropdown.Item>
```
There are 3 special functions in js: bind, call, apply. Here we may get benefit by usage of bind function. It would be very informative and helpful for you to read about all of them and then apply bind function here.

eugenmolchanov commented on frontend/src/pages/warehouse/Warehouses.js:
```diff
@@ -0,0 +1,185 @@
+        <tr key={warehouse.id}>
+            <td>{warehouse.identifier}</td>
+            <td>{warehouse.type}</td>
+            <td>{warehouse.addressDto.city}, {warehouse.addressDto.addressLine1},
```
We may create a function to return the total address cell content. It will receive a parameter addressDto and return whatever we want.

eugenmolchanov commented on src/main/java/by/itech/lab/supplier/controller/UserController.java:
```diff
@@ -42,9 +44,14 @@
-        return userService.getAllDispatchers(customerId, pageable);
+    @GetMapping(URL_DISPATCHERS)
+    public List<UserDto> getListByUsername(@RequestParam String username) {
+        return userService.findListByDispatcherUsername(username);
```
findUsersByDispatcherUsername

alexeyoverchenko replied:
got it

eugenmolchanov commented on src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java:
```diff
@@ -16,8 +16,8 @@
-    @Query("update Warehouse set deletedAt = current_timestamp where id = :id")
-    void delete(@Param("id") Long id);
+    @Query("update Warehouse set deletedAt = current_timestamp where id in :id")
+    void delete(@Param("id") List<Long> id);
```
deleteByIds

alexeyoverchenko replied:
got it

eugenmolchanov commented on src/main/java/by/itech/lab/supplier/service/StateService.java:
```diff
@@ -4,6 +4,11 @@
-public interface StateService {
+import java.util.List;
+
+public interface StateService{
```
Is that application state? Clarify what the state is in the name.

alexeyoverchenko replied:
we think about it

eugenmolchanov commented on src/main/java/by/itech/lab/supplier/service/WarehouseService.java:
```diff
@@ -11,7 +11,13 @@
+
+    WarehouseDto save(final WarehouseDto warehouseDto);
+
+    WarehouseDto findById(final Long warehouseId);
```
return Optional

eugenmolchanov commented on src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:
```diff
@@ -102,20 +102,22 @@ public WarehouseDto save(final WarehouseDto warehouseDto) {
         warehouseMapper.map(warehouseDto, warehouse);
-        return warehouseRepository.save(warehouse);
+        Warehouse saved = warehouseRepository.save(warehouse);
+        System.out.println(warehouseDto.getDispatchersId());
```
Remove this line.

alexeyoverchenko replied:
done

eugenmolchanov reviewed (changes requested):
Please, see the comments.

alexeyoverchenko committed:
- added functionality for decoupling the dispatcher from the warehouse during deletion
- new validation connecting with "type" changes (different conditions for factory and warehouse) also for unique "identifier"

alexeyoverchenko committed:
- fixed asynchronous bugs

alexeyoverchenko committed:
- after cleaning

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/App.js
#	src/main/java/by/itech/lab/supplier/controller/UserController.java
#	src/main/java/by/itech/lab/supplier/controller/pre_tests.http

alexeyoverchenko committed:
- last version

alexeyoverchenko committed:
Merge remote-tracking branch 'origin/sup-46_warehouse_page' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/pages/warehouse/ModalAddWarehouse.js
#	src/main/java/by/itech/lab/supplier/controller/UsStateController.java
#	src/main/java/by/itech/lab/supplier/controller/WarehouseController.java
#	src/main/java/by/itech/lab/supplier/domain/Address.java
#	src/main/java/by/itech/lab/supplier/domain/Tax.java
#	src/main/java/by/itech/lab/supplier/dto/mapper/AddressMapper.java
#	src/main/java/by/itech/lab/supplier/dto/mapper/TaxMapper.java
#	src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java
#	src/main/java/by/itech/lab/supplier/service/WarehouseService.java
#	src/main/java/by/itech/lab/supplier/service/impl/UsStateServiceImpl.java

alexeyoverchenko committed:
Merge branch 'develop' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/App.js
#	src/main/java/by/itech/lab/supplier/controller/UserController.java
#	src/main/java/by/itech/lab/supplier/controller/pre_tests.http

alexeyoverchenko committed:
Merge remote-tracking branch 'origin/sup-46_warehouse_page' into sup-46_warehouse_page

# Conflicts:
#	frontend/src/App.js
#	src/main/java/by/itech/lab/supplier/controller/UserController.java
#	src/main/java/by/itech/lab/supplier/controller/pre_tests.http
